src/UKHO.S100PermitService.StubService/Stubs/UserPermitsServiceStub.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Net;
+using UKHO.S100PermitService.StubService.Configuration;
+using WireMock;
+using WireMock.Matchers;
+using WireMock.RequestBuilders;
+using WireMock.ResponseBuilders;
+using WireMock.Server;
+using WireMock.Types;
+using WireMock.Util;
+
+namespace UKHO.S100PermitService.StubService.Stubs
+{
+    public class UserPermitsServiceStub : IStub
+    {
+        private readonly UserPermitsServiceConfiguration _userPermitsServiceConfiguration;
+        private const string ApplicationType = "application/json";
+        private const string ResponseFileDirectory = @"StubData\UserPermits";
+        private readonly string _responseFileDirectoryPath = Path.Combine(Environment.CurrentDirectory, ResponseFileDirectory);
+
+        public UserPermitsServiceStub(UserPermitsServiceConfiguration userPermitsServiceConfiguration)
+        {
+            _userPermitsServiceConfiguration = userPermitsServiceConfiguration ?? throw new ArgumentNullException(nameof(userPermitsServiceConfiguration));
+        }
+
+        public void ConfigureStub(WireMockServer server)
+        {
+            server
+                .Given(Request.Create()
+                .WithPath(new WildcardMatcher(_userPermitsServiceConfiguration.Url + "/*"))
+                .UsingGet()
+                .WithHeader("Authorization", "Bearer ", MatchBehaviour.RejectOnMatch))
+                .RespondWith(Response.Create()
+                .WithStatusCode(HttpStatusCode.Unauthorized)
+                .WithHeader("X-Correlation-ID", Guid.NewGuid().ToString())
+                .WithBodyFromFile(Path.Combine(_responseFileDirectoryPath, "response-401.json")));
+
+            server
+                .Given(Request.Create()
+                .WithPath(new WildcardMatcher(_userPermitsServiceConfiguration.Url + "/*"))
+                .UsingGet()
+                .WithHeader("Authorization", "Bearer *", MatchBehaviour.AcceptOnMatch))
+                .RespondWith(Response.Create()
+                .WithCallback(SetResponseFromLicenseId));
+        }
+
+        private ResponseMessage SetResponseFromLicenseId(IRequestMessage request)
```

Comment:
Can we rename the variable  request as requestMessage
```suggestion
        private ResponseMessage SetResponseFromLicenseId(IRequestMessage requestMessage)
```

Reply:
Done